modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/components/FieldSelectModalContent.tsx

```diff
@@ -298,36 +312,51 @@ const FieldSelectModalContent = ({
 									query,
 								}: IFieldTreeItem) => (
 									<TreeView.Item>
-										<TreeView.ItemStack
-											disabled={
-												initialChildren &&
-												!!initialChildren.length
-													? true
-													: false
-											}
-											expanderDisabled={false}
-										>
-											<ClayCheckbox checked>
-												<span className="font-weight-normal pl-1 text-3">
+										{selectionMode === 'single' ? (
+											<TreeView.ItemStack
+												disabled={
+													!!initialChildren?.length
+												}
+												expanderDisabled={false}
+											>
+												<ClayCheckbox checked />
+
+												<Highlight
+													query={query}
+													text={label}
+												/>
+											</TreeView.ItemStack>
+										) : (
+											<TreeView.ItemStack>
+												<ClayCheckbox checked>
 													<Highlight
 														query={query}
 														text={label}
 													/>
-												</span>
-											</ClayCheckbox>
-										</TreeView.ItemStack>
+												</ClayCheckbox>
+											</TreeView.ItemStack>
+										)}
 
 										<TreeView.Group items={children}>
 											{({label}: IFieldTreeItem) => (
 												<TreeView.Item>
 													<ClayCheckbox checked>
-														<span className="font-weight-normal pl-1 text-3">
+														{selectionMode ===
+															'multiple' && (
 															<Highlight
 																query={query}
 																text={label}
 															/>
-														</span>
+														)}
 													</ClayCheckbox>
+
+													{selectionMode ===
+														'single' && (
+														<Highlight
+															query={query}
+															text={label}
+														/>
+													)}
```

Comment:
I tried to extract this to buy some space, but `TreeView` really doesn't like it. I left it in, rather than diving deep in Clay implementation.